code/modules/clothing/chameleon/chameleon_gun.dm

```diff
@@ -54,36 +55,33 @@
 	inhand_x_dimension = gun_to_set.inhand_x_dimension
 	inhand_y_dimension = gun_to_set.inhand_y_dimension
 
+	// We dupe this casing and then delete it at the end, to grab the projectile.
+	var/obj/item/ammo_casing/casing_to_dupe
+
 	if(istype(gun_to_set, /obj/item/gun/ballistic))
 		var/obj/item/gun/ballistic/ball_gun = gun_to_set
-		var/obj/item/ammo_box/ball_ammo = new ball_gun.spawn_magazine_type(gun_to_set)
-		qdel(ball_gun)
-
-		if(!istype(ball_ammo) || !ball_ammo.ammo_type)
-			qdel(ball_ammo)
-			return FALSE
-
-		var/obj/item/ammo_casing/ball_cartridge = new ball_ammo.ammo_type(gun_to_set)
-		set_chameleon_ammo(ball_cartridge)
+		// We also need to copy the starting magazine for ballistics.
+		casing_to_dupe = initial(ball_gun.spawn_magazine_type.ammo_type)
+		casing_to_dupe = new casing_to_dupe
 
 	else if(istype(gun_to_set, /obj/item/gun/magic))
 		var/obj/item/gun/magic/magic_gun = gun_to_set
-		var/obj/item/ammo_casing/magic_cartridge = new magic_gun.ammo_type(gun_to_set)
-		set_chameleon_ammo(magic_cartridge)
+		casing_to_dupe = new magic_gun.ammo_type(src)
 
 	else if(istype(gun_to_set, /obj/item/gun/energy))
 		var/obj/item/gun/energy/energy_gun = gun_to_set
+		// Even if the energy gun has multiple ammo types, we copy the first. Energy guns always (should) have a list in ammo_type.
 		if(islist(energy_gun.ammo_type) && energy_gun.ammo_type.len)
-			var/obj/item/ammo_casing/energy_cartridge = energy_gun.ammo_type[1]
-			set_chameleon_ammo(energy_cartridge)
+			casing_to_dupe = energy_gun.ammo_type[1]
```

Comment:
This is still setting the casing to a typepath, not an object, isn't it?

Reply:
It isn't, i tried doing new on it and it blurted out an error of newing an object

Reply:
then it should be making a copy, because it qdels it afterwards